solo-chains/runtime/starlight/src/tests/core_scheduling_tests.rs

```diff
@@ -0,0 +1,438 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+#![cfg(test)]
+
+use {
+    crate::common::*,
+    cumulus_primitives_core::relay_chain::{vstaging::SchedulerParams, AsyncBackingParams},
+    frame_support::assert_ok,
+    frame_system::pallet_prelude::BlockNumberFor,
+    primitives::runtime_api::runtime_decl_for_parachain_host::ParachainHostV11,
+    runtime_parachains::paras::{ParaGenesisArgs, ParaKind},
+    sp_keystore::testing::MemoryKeystore,
+    sp_std::{collections::btree_map::BTreeMap, vec},
+    starlight_runtime::{ContainerRegistrar, OnDemandAssignmentProvider, Paras, ParasSudoWrapper},
+    std::sync::Arc,
+    tp_traits::SlotFrequency,
+};
+
+mod common;
+
+const UNIT: Balance = 1_000_000_000_000_000_000;
+
```

Comment:
```suggestion
```

Also add `mod core_scheduling_tests` alongside `mod session_keys` in tests/mod.rs